central/networkgraph/service/service_impl.go

```diff
@@ -311,9 +314,67 @@ func (s *serviceImpl) getNetworkGraph(ctx context.Context, request *v1.NetworkGr
 			node.QueryMatch = true
 		}
 	}
+
+	if request.IncludePolicies {
+		err := s.enhanceWithNetworkPolicyIsolationInfo(ctx, graph)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return graph, nil
 }
 
+func (s *serviceImpl) enhanceWithNetworkPolicyIsolationInfo(ctx context.Context, graph *v1.NetworkGraph) error {
+	var deploymentIds []string
+	for _, node := range graph.GetNodes() {
+		if node.GetEntity().GetType() == storage.NetworkEntityInfo_DEPLOYMENT {
+			deploymentIds = append(deploymentIds, node.GetEntity().GetId())
+		}
+	}
+
+	deploymentObjects, err := s.deployments.GetDeployments(ctx, deploymentIds)
+	if err != nil {
+		return err
+	}
+
+	clusterNamespaceContext := set.NewSet[deploymentMatcher.ClusterNamespace]()
+	for _, deployment := range deploymentObjects {
+		clusterNamespaceContext.Add(deploymentMatcher.ClusterNamespace{
+			Cluster:   deployment.GetClusterId(),
+			Namespace: deployment.GetNamespace(),
+		})
+	}
+
+	matcher, err := deploymentMatcher.BuildMatcher(ctx, s.networkPolicy, clusterNamespaceContext.AsSlice())
+	if err != nil {
+		return err
+	}
+
+	isolationDetails := make(map[string]deploymentMatcher.IsolationDetails, len(deploymentObjects))
+	for _, deployment := range deploymentObjects {
+		details := matcher.GetIsolationDetails(deployment)
+		isolationDetails[deployment.GetId()] = deploymentMatcher.IsolationDetails{
+			IngressIsolated: details.IngressIsolated,
+			EgressIsolated:  details.EgressIsolated,
+			PolicyIDs:       details.PolicyIDs,
+		}
+	}
+
+	for idx, node := range graph.Nodes {
+		if node.GetEntity().GetType() == storage.NetworkEntityInfo_DEPLOYMENT {
+			deploymentID := node.GetEntity().GetId()
+			if isolationDetail, ok := isolationDetails[deploymentID]; ok {
+				graph.Nodes[idx].NonIsolatedEgress = !isolationDetail.EgressIsolated
```

Comment:
nodes are pointers so no need to nodes[idx]

Reply:
Done